### ISSUE TITLE
extract rel-me/h-card stuff to an external plugin

### ISSUE THREAD

**Issue by pfefferle** [indieweb/wordpress-indieweb#40]
I am not sure if the Rel-Me/h-card header/widget stuff should be part of the Indie plugin core. It adds a lot of weird stuff to my header:

``` html
<link rel="me" href="https://twitter.com/@pfefferle" />
<link rel="me" href="https://www.facebook.com/http://facebook.com/pfefferle" />
```

because there are other plugins that extend the form fields and perhaps use a different syntax.

Atm I can't remove these settings without coding.

**Comment by pfefferle** [comment]
It also breaks my author-urls because it seems to overwrite them with the URL of the profile.

**Comment by pfefferle** [comment]
There is also no way to disable the "Single Author Site" setting if it was once enabled.

**Comment by pfefferle** [comment]
Just found https://github.com/indieweb/wordpress-indieweb/issues/30

**Comment by dshanske** [comment]
I think the issue of addressing the behavioral issues is different. Research with people at HWCs and IWCs has confirmed that having the plugin establish your identity(h-card and rel-me enhancements) gives a clear picture of why to install it and helps move people forward.

I'm going to consider having the settings allow for totally disabling it, but it does override the WordPress concept of an opinionated default. The default should be for it to be on regardless. Also, the form field extension issue...we could opt to switch to a prefixed field, which is recommended. But then there would be meta duplication regardless.

**Comment by pfefferle** [comment]
where is the h-card stuff used/displayed atm?

**Comment by pfefferle** [comment]
We should at least add a function to disable the complete feature, otherwise I will not be able to install the plugin.

**Comment by dshanske** [comment]
I am agreeing with that point. As long as it isn't disabled by default, there should be an option to turn it off. The H-Card stuff is still a pending feature. Basically a simple embeddable h-card that can be used either as a shortcode, a widget, etc.

Part of the basis for this was the old WordPress Extended Profile plugin(https://github.com/singpolyma/wordpress-extended-profile) which supports the classic microformats vocabulary. The Indieweb plugin, now having taken on the identity component of IndieMark, should add the extra fields and offer an output option.

**Comment by pfefferle** [comment]
I commited a first version to disable the features by a constant... this would work for me for now...

**Comment by pfefferle** [comment]
But there is still a problem with the general settings it seems... it adds stuff that effects other plugins so either we should move the parts to the corresponding classes and hook them into the settings page or hide them with the same constants...

**Comment by dshanske** [comment]
I am going to deploy some minor changes to cover this and document it tonight. I only got started this morning.

**Comment by chrisaldrich** [comment]
<a href="https://github.com/indieweb/wordpress-indieweb/issues/40" class="u-in-reply-to"></a>I'd had the same original problem pfefferle notes above.  

Note also that some of the fields in the User Profile that are used to add rel-me tags are also used by other plugins occasionally and they're not all done in a standard/canonical way. (Does WP specify if they should be full URLs or just usernames anywhere?) In particular, the popular "All In One SEO" plugin (1 million+ installs, which competes directly with Yoast) and is fairly ubiquitous uses some of these social media fields (typically with full URLs so https://www.twitter.com/username instead of @username or even username) to add meta-data in their plugin. 

It may be better to include all these accounts separately as their own data fields within a freestanding plugin so there isn't any conflict with other plugins that may not use them canonically.

**Comment by dshanske** [comment]
Well, the issue goes to the fact that removing them eliminates the functionality. Creating another plugin is just adding to the problem. The simplest solution is to use a prefix in the meta key. Instead of twitter, which is used inconsistently, try iw_twitter. We had this originally, but it seemed that most plugins were being consistent, but not all. The potential downside is duplication of fields by different plugins. But it is better than inconsistent uses. 

The simplicity of the current implementation in onboarding needs to be maintained.

**Comment by pfefferle** [comment]
I would prefer using `indieweb_` as prefix for everything (function prefixed, db prefixes, ...), because it is not that long and it is more descriptive for someone looking at the db and trying to find out, what db settings are used by which plugin.

**Comment by pfefferle** [comment]
I would also prefer to not use different fields, because it is even worse to explain to a user why there are several Facebook fields, that have to be filled seperately. I would prefer a more intelligent handling, for example like we did in the OpenGraph plugin: https://github.com/willnorris/wordpress-opengraph/blob/master/opengraph.php#L369

**Comment by dshanske** [comment]
That looks like it would work.

**Comment by chrisaldrich** [comment]
<a href="https://github.com/indieweb/wordpress-indieweb/issues/40" class="u-in-reply-to"></a>As a supplementary note, JetPack v4.5 now includes a Gravatar widget which functions a lot like the H-card Widget and pulls in data from the user's profile as well as their Gravatar account. Now, if only it had microformats... 

Since the code and UI may be worth looking at: https://github.com/Automattic/jetpack/tree/master/modules/widgets

Announcement:  https://jetpack.com/2017/01/17/jetpack-4-5-monetize-your-site-brand-new-videopress-and-many-many-new-shortcodes/

**Comment by pfefferle** [comment]
We have to really make change all the IndieWeb settings to an opt-in feature... After activating the IndieWeb plugin it breaks a lot of other things, so I am not able to use the plugin atm. I am not sure wich other side-effects the plugin causes, when it replaces the author URL. I am using this feature very heavily in other plugins and can't deactivate that feature.

**Comment by dshanske** [comment]
Addressed in PR #52

**Comment by dshanske** [comment]
Closing this for now based on issues addressed.
